alpine.dockerfile

```diff
@@ -25,49 +25,29 @@ LABEL maintainer="Microsoft" \
       org.label-schema.vcs-url="https://github.com/Azure/azure-cli.git" \
       org.label-schema.docker.cmd="docker run -v \${HOME}/.azure:/root/.azure -it mcr.microsoft.com/azure-cli:$CLI_VERSION"
 
-# bash gcc make openssl-dev libffi-dev musl-dev - dependencies required for CLI
-# openssh - included for ssh-keygen
-# ca-certificates
 
-# curl - required for installing jp
-# jq - we include jq as a useful tool
-# pip wheel - required for CLI packaging
-# jmespath-terminal - we include jpterm as a useful tool
-# libintl and icu-libs - required by azure devops artifact (az extension add --name azure-devops)
-
-# We don't use openssl (3.0) for now. We only install it so that users can use it.
-RUN apk add --no-cache bash openssh ca-certificates jq curl openssl perl git zip \
- && apk add --no-cache --virtual .build-deps gcc make openssl-dev libffi-dev musl-dev linux-headers \
```

Comment:
We used to remove `.build-deps` until https://github.com/Azure/azure-cli/pull/8291.
PS: `az extension add -n azure-batch-cli-extensions` works without those deps, I guess some dependencies provides the wheel packages as time goes by.

It's not a good idea to choose Aline as based image, some package does not provide the binary package and the user has to build from source, for example, `psutil` package.